lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -139,15 +180,47 @@ def conditional?(node)
         end
 
         def last_call_of_method?(node)
-          node.parent && node.parent.children.size == node.sibling_index + 1
+          return false unless cop_config['AllowImplicitReturn']
+          node.parent &&
+            (node.parent.def_type? || node.parent.block_type?) &&
+            node.parent.children.size == node.sibling_index + 1
+        end
+
+        def argument?(node)
+          # positional argument
+          return true if node.argument?
+          # keyword argument
+          node.parent && node.parent.parent &&
```

Comment:
Btw, seem here we can avoid a lot of repetition by assigning `node.parent.parent` to some variable.